palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/service/SimplePalisadeService.java

```diff
@@ -75,41 +79,118 @@ public SimplePalisadeService(final AuditService auditService, final UserService
+        resourceRequest.setOriginalRequestId(request.getId());
+        return resourceService
+                .getResourcesById(resourceRequest)
+                .exceptionally((Throwable ex) -> {
+                    auditRequestReceivedException(request, ex, ResourceService.class);
+                    throw new RegisterRequestException("Exception from resourceService", ex);
+                });
+    }
+
+    private CompletableFuture<Map<LeafResource, Rules>> getFutureResourceRules(
+            final RegisterDataRequest request,
+            final CompletableFuture<User> futureUser,
+            final CompletableFuture<Set<LeafResource>> futureResources) {
+
+        return futureUser.thenCombine(futureResources, (User user, Collection<LeafResource> resources) -> {
+            final GetPolicyRequest policyRequest = new GetPolicyRequest()
+                    .user(user)
+                    .context(request.getContext())
+                    .resources(resources);
+            policyRequest.setOriginalRequestId(request.getId());
+            return policyService.getPolicy(policyRequest).join();
+        }).exceptionally((Throwable ex) -> {
+            auditRequestReceivedException(request, ex, PolicyService.class);
+            throw new RegisterRequestException("Exception from policyService", ex);
+        });
+    }
+
+    @Override
+    public DataRequestResponse registerDataRequest(final RegisterDataRequest request) {
+        requireNonNull(request, "request");
+        try {
+            LOGGER.debug("Registering data request: {}", request);
+            LOGGER.info("Registering data request: {}", request.getId());
+
+            CompletableFuture<User> futureUser = getFutureUser(request);
+            CompletableFuture<Set<LeafResource>> futureResources = getFutureResources(request);
+            CompletableFuture<Map<LeafResource, Rules>> futureRules = getFutureResourceRules(request, futureUser, futureResources);
+
+            final String token = request.getUserId().getId() + "-" + UUID.randomUUID().toString();
+            LOGGER.info("Assigned token {} for request {}", token, request.getId());
+            CompletableFuture<DataRequestResponse> futureResponse = CompletableFuture.supplyAsync(
+                    () -> aggregationService.aggregateDataRequestResults(
+                            request,
+                            futureUser,
+                            futureResources,
+                            futureRules,
+                            token),
+                    executor);
+
+            DataRequestResponse response = futureResponse.join();
+            auditRequestComplete(request, futureUser.join(), futureResources.join(), request.getContext());
+
+            return response;
+        } catch (RuntimeException ex) {
+            auditRequestReceivedException(request, ex, SimplePalisadeService.class);
+            throw new RegisterRequestException(ex);
+        } catch (Error err) {
```

Comment:
SQ doesn't like the catching of an Error. This should either be changed to an Exception or ignore the rule within SQ.

Reply:
fixed within SQ